apps/dc_tools/odc/apps/dc_tools/sqs_to_dc.py

```diff
@@ -171,25 +171,36 @@ def do_index_update_dataset(
     dc: Datacube,
     doc2ds: Doc2Dataset,
     update=False,
+    update_if_exists=False,
     allow_unsafe=False,
 ):
     if uri is not None:
+        # Make sure we can create a dataset first
         try:
             ds, err = doc2ds(metadata, uri)
         except ValueError as e:
             raise SQStoDCException(
                 f"Exception thrown when trying to create dataset: '{e}'\n The URI was {uri}"
             )
+
+        # Now do something with the dataset
         if ds is not None:
-            if update:
-                updates = {}
-                if allow_unsafe:
-                    updates = {tuple(): changes.allow_any}
-                dc.index.datasets.update(ds, updates_allowed=updates)
-            else:
-                if dc.index.datasets.has(metadata.get("id")):
+            if dc.index.datasets.has(metadata.get("id")):
```

Comment:
should read `id` from `ds` not from `metadata` dictionary. So `ds.id` rather than `metadata.get("id")`